blog/2024-03-04-config-entry-reconfigure-step.md

```diff
@@ -0,0 +1,44 @@
+---
+author: G Johansson
+authorURL: https://github.com/gjohansson-ST
+title: "Config Entries can now provide a reconfigure step"
+---
+
+As of Home Assistant Core 2024.4 config entries can now be reconfigured by adding a `reconfigure` step in their config flows.
+
+This is not to replace optional configuration (`OptionsFlow`) but instead to allow the user to change setup configuration after a config entry has been created.
+
+### Integrations updating existing config entries
+
+Some integrations implement a reconfiguration option by updating an existing config entry if a matching config entry is being created by the user. This is no longer allowed, and integrations should be migrated to implement the `reconfigure` step.
```

Comment:
> This is no longer allowed

Why is that? Is there a problem with it or reasoning against it?

Reply:
I think it's a logical distinction for the user when configuring a new vs. reconfiguring an existing. 

I believe it can be quite confusing for a user to create a new integration but it results in changing an existing and the user would not know what has changed. 

The parameter needs to be kept for discovery flows so it's not breaking as such but we shouldn't encourage devs to not use it for user-initiated flows.